### PR TITLE
Add TETyper

### PR DESCRIPTION
**Pull request by dfornika**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/tetyper/tetyper.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+<tool id="tetyper" name="TETyper" version="1.1+galaxy0">
```
Comment (bgruening) [inline tools/tetyper/tetyper.xml]: do you want to macrofy the version? No extra file is needed, you could do this inline.

File: tools/tetyper/tetyper.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        <param name="min_qual" type="integer" min="0" value="10" max="100" label="Minimum quality" help="Minimum quality value across extracted flanking sequence." />
+        <param name="snp_profiles" type="data" format="text" label="SNP Profiles" optional="true" />
+        <param name="struct_profiles" type="data" format="text" label="Structural Variant Profiles" optional="true" />
+    </inputs>
+    <outputs>
+        <data name="summary" format="tabular" from_work_dir="output_summary.txt" label="${tool.name} on ${on_string}: summary"/>
+        <data name="snps" format="vcf" from_work_dir="output.vcf" label="${tool.name} on ${on_string}: SNPs"/>
+        <data name="blast" format="tabular" from_work_dir="output_blast.txt" label="${tool.name} on ${on_string}: BLAST alignment"/>
+        <data name="alignment" format="bam" from_work_dir="output.bam" label="${tool.name} on ${on_string}: BWA alignment"/>
+        <data name="log" format="text" from_work_dir="output.log" label="${tool.name} on ${on_string}: log"/>
```
Comment (bgruening) [inline tools/tetyper/tetyper.xml]: should the log file be optional?

File: tools/tetyper/tetyper.xml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        <param name="reference" type="data" format="fasta" label="Transposable Element Reference"/>
+        <param name="flank_length" type="integer" min="4" value="5" max="16" label="Flank Length" help="Length of flanking region to extract."/>
+        <param name="min_reads" type="integer" min="1" value="10" max="100" label="Minimum Reads" help="Minimum read number for including a specific flanking sequence."/>
+        <param name="min_each_strand" type="integer" min="1" value="1" max="100" label="Minimum Reads (each strand)" help="Minimum read number for each strand for including a specific flanking sequence."/>
+        <param name="min_mapped_len" type="integer" min="8" value="30" max="100" label="Minimum Mapped Length" help="Minimum length of mapping for a read to be used in determining flanking sequences. Higher values are more robust to spurious mapping. Lower values will recover more reads."/>
+        <param name="min_qual" type="integer" min="0" value="10" max="100" label="Minimum quality" help="Minimum quality value across extracted flanking sequence." />
+        <param name="snp_profiles" type="data" format="text" label="SNP Profiles" optional="true" />
```
Comment (bgruening) [inline tools/tetyper/tetyper.xml]: can you describe how those files should look like?

Comment (dfornika) [reply tools/tetyper/tetyper.xml]: I've added descriptions to the `<help>` section

File: tools/tetyper/tetyper.xml (hunk 1 above)
Comment (dfornika) [reply tools/tetyper/tetyper.xml]: Done.

File: tools/tetyper/tetyper.xml (hunk 2 above)
Comment (dfornika) [reply tools/tetyper/tetyper.xml]: Sure, but I'm not sure how to do that. I've looked at the Tool XML guide and a few other tools that provide a log output but haven't seen an example.

Could you recommend how to do it? Or point to another tool that I could follow?

Comment (bgruening) [reply tools/tetyper/tetyper.xml]: Add a boolean and ask the user if the log file is needed and than add a filter to the output.

**Comment by bgruening** [comment]
Cool thanks. Please remove the WIP when you think it is ready.

**Pull request merged**